test/k8sT/Services.go

```diff
@@ -579,6 +579,59 @@ var _ = Describe("K8sServicesTest", func() {
 				testCurlFromPods(echoPodLabel, url, 5, 0)
 			}
 		})
+
+		curlClusterIPFromExternalHost := func(shouldFail bool) {
+			clusterIP, _, err := kubectl.GetServiceHostPort(helpers.DefaultNamespace, serviceName)
+			Expect(err).Should(BeNil(), "Cannot get service %s", serviceName)
+			Expect(govalidator.IsIP(clusterIP)).Should(BeTrue(), "ClusterIP is not an IP")
+			httpSVCURL := fmt.Sprintf("http://%s/", net.JoinHostPort(clusterIP, "80"))
+
+			By("testing external connectivity via cluster IP %s", clusterIP)
+
+			status := kubectl.ExecInHostNetNS(context.TODO(), k8s1NodeName, helpers.CurlFail(httpSVCURL))
+			Expect(status).Should(helpers.CMDSuccess(), "cannot curl to service IP from host: %s", status.CombineOutput())
+
+			status = kubectl.ExecInHostNetNS(context.TODO(), outsideNodeName, helpers.CurlFail(httpSVCURL))
+			if shouldFail {
+				Expect(status).ShouldNot(helpers.CMDSuccess(),
+					"%s host unexpectedly connected to service %q, it should fail", outsideNodeName, httpSVCURL)
+			} else {
+				Expect(status).Should(helpers.CMDSuccess(),
+					"cannot curl ClusterIP from external host: %s", status.CombineOutput())
+			}
+		}
```

Comment:
May I suggest a readability change: 

I think removing the bool `shouldFail`from `curlClusterIPFromExternalHost()` and leaving the specific assertions to be done inside the actual `It()` blocks instead. This way, you can still have `curlClusterIPFromExternalHost()` defined which performs the same logic to test, but instead returns `status`. Then the `It()` blocks should simply be:

```go
It /*ThatShouldSucceed*/ (func() {
  Expect(curlClusterIPFromExternalHost()).Should(helpers.CMDSuccess())
})

It /*ThatShouldFail*/ (func() {
  Expect(curlClusterIPFromExternalHost()).Should(helpers.CMDFailure())
}).
```

The reason is to avoid having to do assertions inside helper functions because the stacktrace can be hard to read and also it is easier to reason about this way. What do you think?

Reply:
Yep, totally agreed. Changed.